packages/ui/src/components/ConfirmationCode/ConfirmationCode.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { PatternInput } from "@/components/PatternInput";
+import { isNaN, range } from "lodash";
+import { useToast } from "@/components/Toast";
+import { useFormatMessage } from "@/hooks";
+
+export const ConfirmationCode: React.FC<{
+  /**
+   * validation occurs once the last number written when true
+   * validation occurs on clicking on button when false
+   */
+  autoValidate?: boolean;
+  confirmCode: (newCode: number, auto: boolean) => void;
+}> = ({ autoValidate = false, confirmCode }) => {
+  const toast = useToast();
+  const intl = useFormatMessage();
+
+  const [newCode, setNewCode] = useState("");
+  const inputRefs = useRef<(HTMLInputElement | null)[]>([]);
+
+  inputRefs.current = useMemo(
+    () => range(5).map((_, index) => inputRefs.current[index] ?? null),
+    []
+  );
+
+  const handleInputChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>, index: number) => {
+      const input = e.target;
+      const code = [...newCode];
+      code[index] = input.value;
+      setNewCode(code.join(""));
+    },
+    [newCode]
+  );
+
+  // Handle backspace key
+  const handleBackspace = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "Backspace") inputRefs.current[newCode.length - 1]?.focus();
+    },
+    [newCode]
+  );
+
```

Comment:
There is a package that we use to handle shortcuts. It has a hook called `useHotKey` ([ref](https://github.com/litespace-org/litespace/blob/2a009de803c77110ec123c180535d558db17f68f/apps/web/src/hooks/logger.ts#L28-L33))